Vagrantfile

```diff
@@ -1,93 +1,166 @@
-      curl -fsSL https://pkgs.k8s.io/core:/stable:/v1.30/deb/Release.key | sudo gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
-      echo 'deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.30/deb/ /' | sudo tee /etc/apt/sources.list.d/kubernetes.list
-      sudo apt update
-      sudo apt install -y kubelet kubeadm kubectl
-      sudo apt-mark hold kubelet kubeadm kubectl
-      sudo systemctl enable --now kubelet
-      sudo containerd config default | sudo tee /etc/containerd/config.toml
-      sudo sed -i 's/            SystemdCgroup = false/            SystemdCgroup = true/' /etc/containerd/config.toml
-      sudo sed -i 's|sandbox_image = "registry.k8s.io/pause:3.8"|sandbox_image = "registry.k8s.io/pause:3.9"|g' /etc/containerd/config.toml
-      sudo systemctl restart containerd
-      SHELL
-  end
+        # Create Cluster Init Script:
+        echo "#!/bin/bash"                                                     >  cluster_init.sh
+        echo "echo 'Pulling k8s Images'"                                       >> cluster_init.sh
+        echo "sudo kubeadm config images pull"                                 >> cluster_init.sh
+        echo "echo ''"                                                         >> cluster_init.sh
+        echo "echo 'Initializing Cluster'"                                     >> cluster_init.sh
+        echo "sudo kubeadm init --pod-network-cidr=${BASE_CIDR}/16 --apiserver-advertise-address=${API_SERVER_IP}" >> cluster_init.sh
+        echo "if [ -f /etc/kubernetes/admin.conf ] ; then"                     >> cluster_init.sh
+        echo "  echo ''"                                                       >> cluster_init.sh
+        echo "  echo 'Create local .kube/config'"                              >> cluster_init.sh
+        echo "  mkdir -p \\\${HOME}/.kube"                                     >> cluster_init.sh
+        echo "  sudo cp -i /etc/kubernetes/admin.conf \\\${HOME}/.kube/config" >> cluster_init.sh
+        echo "  sudo chown \\\$(id -u):\\\$(id -g) \\\${HOME}/.kube/config"    >> cluster_init.sh
+        echo "  echo ''"                                                       >> cluster_init.sh
+        echo "  echo 'Install Weave'"                                          >> cluster_init.sh
+        echo "  kubectl apply -f https://github.com/weaveworks/weave/releases/download/v2.8.1/weave-daemonset-k8s.yaml" >> cluster_init.sh
+        echo "fi"                                                              >> cluster_init.sh
+        chmod a+rx cluster_init.sh
 
-  # Define worker VM
-  config.vm.define "worker" do |worker|
-    worker.vm.box = "bento/ubuntu-22.04"
-    worker.vm.network "private_network", ip: "192.168.63.2"
-    worker.vm.hostname = "worker"
-    worker.vm.provider "virtualbox" do |v|
-      v.name = "worker"
-      v.memory = 2048
-      v.cpus = 2
+        # Show the k8s join command:
```

Comment:
Well, I was already creating one script, so I created this one also, even though it could have just as easily been part of the repo and then put in place with a `vagrant scp X cplane` command.